src/marks/raster.d.ts

```diff
@@ -12,43 +29,196 @@ export type RasterInterpolateFunction = (
   V: any[]
 ) => any[];
 
+/** A spatial interpolation method. */
 export type RasterInterpolate = RasterInterpolateName | RasterInterpolateFunction;
 
+/**
+ * A source of pseudo-random numbers in [0, 1). The default source is seeded to
+ * ensure reproducibility.
+ */
 export type RandomSource = () => number;
 
+/**
+ * A sampler function, which returns a value for the given *x* and *y* values in
+ * the current *facet*. For example, to create side-by-side color plots of two
+ * functions of *x* and *y*:
+ *
+ * ```js
+ * Plot.raster({
+ *   fill: (x, y, {fx}) => fx(x) * y,
+ *   fx: [Math.sin, Math.cos],
+ *   x1: 0,
+ *   y1: 0,
+ *   x2: 4 * Math.PI,
+ *   y2: 2,
+ *   interval: 0.2
+ * });
+ * ```
+ */
 export type RasterSampler = (x: number, y: number, facet: number[] & {fx: any; fy: any}) => any;
 
+/** Options for the raster mark. */
 export interface RasterOptions extends Omit<MarkOptions, "fill" | "fillOpacity"> {
+  /** The horizontal coordinate of a sample. */
   x?: ChannelValueSpec;
+  /** The vertical coordinate of a sample. */
   y?: ChannelValueSpec;
+  /**
+   * The lower value of *x*, on the left edge of the raster (right edge if
+   * reversed).
```

Comment:
This and other options need the defaults described (which is a bit tedious to describe because there are a lot of different defaults at play here, but that’s a big part of the value that documentation provides).

Reply:
I hope I got those right.